src/index.d.ts

```diff
@@ -19,6 +19,34 @@ export interface HtmlFormatter extends Formatter {
     hideUnchanged(node?: Element | null, delay?: number): void;
 }
 
+type AddOperation = {
+  opp: string;
+  path: string;
+  value: any
+}
+
+type RemoveOperation = {
+  opp: string;
+  path: string;
+}
+
+type ReplaceOperation = {
+  opp: string;
+  path: string;
+  value: any;
+}
+
+type MoveOperation = {
+  opp: string;
+  path: string;
+}
+
+export type Operation = AddOperation | RemoveOperation | ReplaceOperation | MoveOperation
+
+export interface JSONPatchFormatter {
```

Comment:
Ideally this would inherit from `Formatter`, but we can handle that in a future PR.